pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -2,138 +2,252 @@
+		useMount:            false,
+		includedDevices:     []regexp.Regexp{},
+		excludedDevices:     []regexp.Regexp{},
+		includedFilesystems: []regexp.Regexp{},
+		excludedFilesystems: []regexp.Regexp{},
+		includedMountpoints: []regexp.Regexp{},
+		excludedMountpoints: []regexp.Regexp{},
+		tagByFilesystem:     false,
+		allPartitions:       false,
+		deviceTagRe:         make(map[*regexp.Regexp][]string),
+		includeAllDevices:   true,
+		minDiskSize:         0,
+		tagByLabel:          true,
+		useLsblk:            false,
+		blkidCacheFile:      "",
+		serviceCheckRw:      false,
 	}
+}
 
-	// device mountpoint matches `excluded_mountpoint_re`
-	if c.cfg.excludedMountpointRe != nil && c.cfg.excludedMountpointRe.MatchString(mountpoint) {
-		return true
+func (c *Check) Run() error {
+	sender, err := c.GetSender()
+	if err != nil {
+		return err
 	}
+	if c.cfg.tagByLabel {
+		err = c.fetchAllDeviceLabels()
+		if err != nil {
+			log.Debugf("Unable to fetch device labels: %s", err)
+		}
+	}
+	err = c.collectPartitionMetrics(sender)
+	if err != nil {
+		return err
+	}
+	err = c.collectDiskMetrics(sender)
+	if err != nil {
+		return err
+	}
+	sender.Commit()
 
-	// all good, don't exclude the disk
-	return false
+	return nil
 }
 
-func (c *Check) instanceConfigure(data integration.Data) error {
-	conf := make(map[interface{}]interface{})
-	c.cfg = &diskConfig{}
-	err := yaml.Unmarshal([]byte(data), &conf)
+func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
+	err := c.CommonConfigure(senderManager, initConfig, data, source)
 	if err != nil {
 		return err
 	}
+	return c.diskConfigure(data, initConfig)
+}
 
-	useMount, found := conf["use_mount"]
-	if useMount, ok := useMount.(bool); found && ok {
-		c.cfg.useMount = useMount
+func (c *Check) diskConfigure(data integration.Data, initConfig integration.Data) error {
+	unmarshalledInstanceConfig := make(map[interface{}]interface{})
+	err := yaml.Unmarshal([]byte(data), &unmarshalledInstanceConfig)
+	if err != nil {
+		return err
 	}
-
-	excludedFilesystems, found := conf["excluded_filesystems"]
-	if excludedFilesystems, ok := excludedFilesystems.([]string); found && ok {
-		c.cfg.excludedFilesystems = excludedFilesystems
+	unmarshalledInitConfig := make(map[interface{}]interface{})
+	err = yaml.Unmarshal([]byte(initConfig), &unmarshalledInitConfig)
+	if err != nil {
+		return err
 	}
 
-	// Force exclusion of CDROM (iso9660) from disk check
-	c.cfg.excludedFilesystems = append(c.cfg.excludedFilesystems, "iso9660")
-
-	excludedDisks, found := conf["excluded_disks"]
-	if excludedDisks, ok := excludedDisks.([]string); found && ok {
-		c.cfg.excludedDisks = excludedDisks
+	deprecationsInitConf := map[string]string{
+		"file_system_global_blacklist": "file_system_global_exclude",
+		"device_global_blacklist":      "device_global_exclude",
+		"mount_point_global_blacklist": "mount_point_global_exclude",
 	}
```

Comment:
Should we move this to a constant?

Reply:
Declare it as a `var` instead of a constant

Reply:
What is the benefit in this case?

Reply:
Better readability

Reply:
Since it cannot be created as a `const`, which would make a lot of sense, due to limitations of the `Go` language, I think the best option is to keep it as a local variable to the method in which it is used.
Advantages of using it as a local variable:
- Lower memory consumption: It is only created when the function is called and is released when it ends (not critical in this case). 
- Total encapsulation: It is not exposed outside the function, avoiding accidental modifications.
- Ease of testing and maintenance: It keeps the code cleaner and more modular. Its use is close to its declaration (best practices).